scripts/run-targeted-tests.mjs

```diff
@@ -0,0 +1,179 @@
+import execa from 'execa';
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import glob from 'glob';
+import { spawnProcess } from './utils/index.mjs';
+import hotPackageJson from '../package.json';
+
+const CONFIG_PATHS = [
+  '.config',
+  'scripts'
+];
+const IGNORED_PATHS = [
+  '.changelogs',
+  '.codesandbox',
+  '.github',
+  'bin',
+  'resources',
+  '.editorconfig',
+  '.gitattributes',
+  '.gitignore',
+  '.stylelintignore',
+  'CODE_OF_CONDUCT.md',
+  'handsontable-general-terms.pdf',
+  'handsontable-non-commercial-license.pdf',
+  'hot.config.js',
+  'LICENSE.txt',
+  'README.md',
+  'stylelint.config.js',
+  'CHANGELOG.md',
+];
+
+const argv = yargs(hideBin(process.argv))
+  .alias('p', 'pipeline')
+  .number('p')
+  .describe('p', 'Define the pipeline number the tests will be triggered on.' +
+    '\n If HOT was modified or the tests are triggered on a full-test branch:' +
+    '\n - pipeline 1: `production` tests for Handsontable and package tests.' +
+    '\n - pipeline 2: `unit`, `type`, `walkontable` and `e2e` tests for Handsontable.' +
+    '\n\n If HOT was _not_ modified, both pipelines run the other packages\' tests (distributed ~50/50).' +
+    '\n\n If no pipeline is defined, all the test run consecutively.')
+  .argv;
+
+const workspacePackages = hotPackageJson.workspaces.packages;
+const touchedProjects = [];
+
+/**
+ * Distribute the tests between the pipelines as follows:
+ * 1. If HOT was modified or the tests are triggered on a full-test branch:
+ *     - pipeline 1: `production` tests for Handsontable and all package tests.
+ *     - pipeline 2: `unit`, `type`, `walkontable` and `e2e` tests for Handsontable.
+ * 2. If HOT was _not_ modified, both pipelines run the other packages' tests (distributed ~50/50).
+ * 3. If no pipeline is defined, all the test run consecutively.
+ *
+ * @param {string[]} modifiedProjects Array of modified project names.
+ * @returns {Promise<void>}
+ */
+async function distributeBetweenPipelines(modifiedProjects) {
+  const pipelineCount = 2;
+  const pipeline = parseInt(argv.pipeline, 10);
```

Comment:
After [configuring the number parser](https://github.com/handsontable/handsontable/pull/7434/files#diff-6534079d1eab39461e932d9bd3bda6e2463b6859175615d12f2f8a072513b694R34) I think that the manual parsing can be skipped.
```suggestion
  const pipeline = argv.pipeline;
```